sofy documentation (#880)
codemagic-ci-cd/codemagic-docs: content/integrations/sofy-integration.md

---
title: Sofy integration
description: How to integrate your workflows with LambdaTest using codemagic.yaml
weight: 11
---

**Sofy** is a testing automation platform that uses testing capabilities to enable tests without writing a single line of code. 

## Configuring Sofy App Automation with Codemagic

> It is possible to publish binaries generated by **Codemagic** and schedule automation tests with **Sofy** via **codemagic.yaml**. More info can be found [here](../yaml/yaml-getting-started/). In order to achieve it, you need to sign up with [**Sofy**](https://sofy.ai/) where you will access your API Key which is a must to have for deploying your apps via **Codemagic**. 

Reviewer: ```suggestion
It is possible to publish binaries generated by **Codemagic** and schedule automation tests with **Sofy** via **codemagic.yaml**. More info about working with Codemagic's YAML configuration can be found [here](../yaml/yaml-getting-started/). You will need to sign up with [**Sofy**](https://sofy.ai/) so you can obtain an API Key which is required for deploying your apps via **Codemagic**. 
```

## Real Time and App Automation

The following **cURL** commands will help you achieveing the deployment process:

**Uploading apps**

```
- name: Publish APK / AAB / IPA to Sofy
  script: |
    curl --location --request POST 'https://api.sofy.ai/api/AppTests/buildUpload' \
    --header "SubscriptionKey: $SOFY_SUBSCRIPTION_KEY" \

>     --form "applicationFile=@"$APK_PATH""

Reviewer: ```suggestion
    --form "applicationFile=@'$APK_PATH'"
```

>     --form "applicationFile=@'$APK_PATH'"

Author: done

```

**Scheduling automation tests**

```
- name: Schedule an automation test with Sofy
  script: |
   curl --location --request POST 'https://api.sofy.ai/api/CICD/ScheduleAutomatedTestRun' 
    --header 'Content-Type: application/json'    
    --data-raw '{

>     "APIKey":"$SOFY_API_KEY",
>     "ScheduledID":$SCHEDULE_ID

Reviewer: ```suggestion
    "APIKey": "$SOFY_API_KEY",
    "ScheduledID": "$SCHEDULE_ID"
```

    }'
```

**Checking status of scheduled tests**

```
- name: Schedule an automation test with Sofy
  script: |
    curl --location --request POST 'https://api.sofy.ai/api/CICD/ScheduleAutomatedTestRunStatus'    
    --header 'Content-Type: application/json'     
    --data-raw '{

>         "APIKey":"$SOFY_API_KEY",
>         "ScheduledID":$SCHEDULE_ID

Reviewer: ```suggestion
        "APIKey": "$SOFY_API_KEY",
        "ScheduledID": "$SCHEDULE_ID"
```

    }'
```

> **$SOFY_SUBSCRIPTION_KEY**, **ScheduledID** and **$SOFY_API_KEY** are coming from your **Sofy** account settings and can be named to anything preferred. More info can be found [here](https://docs.codemagic.io/variables/environment-variable-groups/) about how to set up environment variable groups with Codemagic. Environment variables can be added in the Codemagic web app using the ‘Environment variables’ tab. You can then and import your variable groups into your codemagic.yaml. For example, if you named your variable group ‘browserstack_credentials’, you would import it as follows:

Reviewer: ```suggestion
**$SOFY_SUBSCRIPTION_KEY**, **ScheduledID** and **$SOFY_API_KEY** are coming from your **Sofy** account settings and can be named to anything you prefer. More info can be found [here](https://docs.codemagic.io/variables/environment-variable-groups/) about how to set up environment variable groups with Codemagic. Environment variables can be added in the Codemagic web app using the ‘Environment variables’ tab. You can then import your variable groups into your codemagic.yaml. For example, if you named your variable group ‘sofy_credentials’, you would import it as follows:
```
